src/libraries/Microsoft.Extensions.Primitives/src/CompositeChangeToken.cs

```diff
@@ -13,11 +14,14 @@ namespace Microsoft.Extensions.Primitives
     /// </summary>
     public class CompositeChangeToken : IChangeToken
     {
-        private static readonly Action<object> _onChangeDelegate = OnChange;
-        private readonly object _callbackLock = new object();
-        private CancellationTokenSource _cancellationTokenSource;
-        private bool _registeredCallbackProxy;
-        private List<IDisposable> _disposables;
+        private static readonly Action<object?> _onChangeDelegate = OnChange;
+        private readonly object _callbackLock = new();
+        private CancellationTokenSource? _cancellationTokenSource;
+        private List<IDisposable>? _disposables;
+
+        [MemberNotNullWhen(true, nameof(_cancellationTokenSource))]
+        [MemberNotNullWhen(true, nameof(_disposables))]
+        private bool _registeredCallbackProxy { get; set; }
```

Comment:
also please re-order back the fields

Reply:
But this is property now.. Do i need to move it back with the fields?

Reply:
Since it is a property, it shouldn't be named like a field.